src/Teacher.js

```diff
@@ -83,6 +83,57 @@ function AssessmentToHTMLStrTeacher(assessment)
     return htmlStr;
 }
 
+function HTMLStrLetterGrade()
+{
+    const studentIds = database.database.GetStudentIDs()
+    const numberOfStudents = studentIds.length
+
+    const studentData = studentIds.map((studentId) =>{
+        return {
+            id: studentId,
+            grade: database.database.GetLetterGrade(studentId)
+        }
+    })
+    console.log(studentData[0])
+    
+    return `
+    <div ><h2>SOEN 287 Section Q </h2> </div>
+    <div><h3>Assign letter grade for the students</h3></div>
+    <div><h4>ID &emsp;&emsp;&emsp;&emsp; Letter Grade</h4></div>
+
+    <div class="mainBox">
+        <form id = "letter_grade">
+        </form>
+    </div>
+   
+    <script>
+        // Generate a dynamic number of inputs
```

Comment:
Not that it is a big issue but ideally this would be in another javascript file that is generated. You could then have the letter grade html code be in it's own file since it does not need to be generated